client/ayon_core/pipeline/workfile/workfile_template_builder.py

```diff
@@ -506,55 +506,61 @@ def build_template(
             keep_placeholders (bool): Add flag to placeholder data for
                 hosts to decide if they want to remove
                 placeholder after it is used.
-            create_first_version (bool): create first version of a workfile
-            workfile_creation_enabled (bool): If True, it might create
-                                              first version but ignore
-                                              process if version is created
+            create_first_version (bool): Create first version of a workfile.
+                 When set to True, this option initiates the saving of the
+                 workfile for an initial version. It will skip saving if
+                 a version already exists.
+            workfile_creation_enabled (bool): Whether we are creating a new
+                workfile. If False, we assume we explicitly want to build the
+                template in our current scene. Otherwise we only build if the
+                current file is not an existing saved workfile but a "new"
+                file.
 
         """
-        if any(
-            value is None
-            for value in [
-                template_path,
-                keep_placeholders,
-                create_first_version,
-            ]
+
+        # Get default values if not provided
+        if (
+                template_path is None
+                or keep_placeholders is None
+                or create_first_version is None
         ):
-            template_preset = self.get_template_preset()
-            if template_path is None:
-                template_path = template_preset["path"]
+            preset = self.get_template_preset()
+            template_path: str = template_path or preset["path"]
             if keep_placeholders is None:
-                keep_placeholders = template_preset["keep_placeholder"]
+                keep_placeholders: bool = preset["keep_placeholder"]
             if create_first_version is None:
-                create_first_version = template_preset["create_first_version"]
-
-        # check if first version is created
-        created_version_workfile = False
-        if create_first_version:
-            created_version_workfile = self.create_first_workfile_version()
+                create_first_version: bool = preset["create_first_version"]
 
-        # if first version is created, import template
-        # and populate placeholders
+        # Build the template if current workfile is a new unsaved file
```

Comment:
this general docstring should cover all the possible cases of the conditionals within. If you read this and after you read the docstring that follows with `# If not a workfile creation, an explicit load template was requested, so we always want to build the template` your head explodes

Reply:
Could you make a suggestion please?

Reply:
Yeah but I think we should wait to resolve this discussion here https://github.com/ynput/ayon-core/pull/871#discussion_r1747851707 first

But I would expect it to be something along these lines if there's two conditionals:
```
Build the template if we are explicitly requesting it or it's an empty scene
```